### DISCUSSION TITLE
Custom Code style formatting

### DISCUSSION THREAD

**Discussion by dodonut** [mfussenegger/nvim-jdtls#187] [Q&A] [answered]
In my work place we have a custom xml code style, and I do not know how to make it work.
I have seen this example config in the repo wiki (which gave me hope), but the using the vim.lsp.buf.formatting_sync() it format in another style.
![image](https://user-images.githubusercontent.com/19748045/142495101-6bd1864f-6fd5-4b60-9071-fcfa9a7a78dd.png)

Does someone know how to help me?

**Comment by mfussenegger** [comment, marked as answer]
Could you try with nested properties?

```
java = {
  format = {
    settings = {
      url = 'path/to..',
      profile = 'GoogleStyle',
    }
  }
}
```

**Reply by dodonut** [reply]
@mfussenegger It did not work. Should I activate somehow? Should I be using vim.lsp.buf.formatting_sync() to apply the code style? Here's my config... everything else works fine (completion, diagnostics, etc.) 
p.s.: the 'formatpath()' func works fine, I've printed the config.settings.java table.
[config.txt](https://github.com/mfussenegger/nvim-jdtls/files/7580653/config.txt)

**Reply by mfussenegger** [reply]
See https://github.com/mfussenegger/nvim-jdtls/issues/203#issuecomment-1006634815

**Comment by dodonut** [comment]
@mfussenegger Looking more into the lsp, I found this on the logs. Same config as the comment above.
![image](https://user-images.githubusercontent.com/19748045/143887152-b10c2fb4-b037-4ede-987d-72f608ca1561.png)

**Comment by spoint-twilio** [comment]
Hey everyone, I am reopening an old discussion but I can't make my custom .xml file to be used.
here is my nvim-jdtls in my neovim config : 
```
local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
local workspace_dir = "/Users/spoint/dev/jdtls_data/" .. project_name
return {
  {
    "mfussenegger/nvim-jdtls",
    opts = {
      -- Ensure jdtls starts with Java 21
      full_cmd = function()
        return {
          "/Users/spoint/.jenv/versions/21/bin/java",
          "-Declipse.application=org.eclipse.jdt.ls.core.id1",
          "-Dosgi.bundles.defaultStartLevel=4",
          "-Declipse.product=org.eclipse.jdt.ls.core.product",
          "-Dlog.protocol=true",
          "-Dlog.level=ALL",
          "-Xmx1g",
          "--add-modules=ALL-SYSTEM",
          "--add-opens",
          "java.base/java.util=ALL-UNNAMED",
          "--add-opens",
          "java.base/java.lang=ALL-UNNAMED",
          "-javaagent:/Users/spoint/.config/local/share/nvim/mason/share/jdtls/lombok.jar",
          "-jar",
          "/Users/spoint/.config/local/share/nvim/mason/packages/jdtls/plugins/org.eclipse.equinox.launcher_1.7.0.v20250519-0528.jar",
          "-configuration",
          "/Users/spoint/.config/local/share/nvim/mason/packages/jdtls/config_mac",
          "-data",
          workspace_dir,
        }
      end,
      settings = {
        java = {
          configuration = {
            runtimes = {
              {
                name = "JavaSE-21",
                path = "/Users/spoint/.jenv/versions/21",
              },
              {
                name = "JavaSE-11",
                path = "/Users/spoint/.jenv/versions/11",
              },
            },
          },
          format = {
            enabled = true,
            settings = {
              url = "/Users/spoint/dotfiles/nvim/.config/nvim/formatter/formatter.xml",
            },
          },
          completion = {
            importOrder = { "java", "javax", "com", "org" },
          },
        },
      },
    },
  },
}
```
and here is the linked formatter.xml file which is for the moment really minimal : 
```
<?xml version="1.0" encoding="UTF-8"?>
<profiles version="12">
  <profile name="TwilioCheckstyle" kind="CodeFormatterProfile">
    <setting id="org.eclipse.jdt.core.formatter.tabulation.size" value="8"/>
    <setting id="org.eclipse.jdt.core.formatter.lineSplit" value="100"/>
    <setting id="org.eclipse.jdt.core.formatter.tabulation.char" value="space"/>
  </profile>
</profiles>
```
I still have only 2 spaces as indentation and lines bigger than 100 characters... it is not respecting the .xml file

**Reply by szymi88** [reply]
I have the same problem, have you managed to resolve this? 
jdtls seem to load my profile because it complains in the log if I don't point it to the existing profile name, but ConformInfo shows:
`Formatters for this buffer:
LSP: jdtls
lsp error: Unknown formatter. Formatter config missing or incomplete`
